libr/reg/reg.c

```diff
@@ -119,7 +122,7 @@ R_API void r_reg_free(RReg *reg) {
 		return;
 
 	for (i = 0; i < R_REG_TYPE_LAST; i++) {
-		r_list_purge (reg->regset[i].pool);
+		r_list_free (reg->regset[i].pool);
```

Comment:
My observation is right after this call, `reg->regset[i].pool` is set to `NULL` making no one pointing to the memory locations, so I thought of using free instead of purge.

I think the reason for using purge instead of free is in `r_reg_arena_push` function
 `r_list_push (reg->regset[i].pool, b);`
 is used to push arena in to pool. So purge was used instead of free since there is a pointer in the list, which was pointed using `reg->regset[i].arena` but since we are any way not freeing any arena while freeing `reg` I think it should be valid.